server/socket-manager.js

```diff
@@ -10,11 +14,33 @@ const authenticateCookie = (socket, callback) => {
 };
 
 const socketManager = (socket) => {
-  socket.on('authenticate', () => {
-    authenticateCookie(socket, (err, data) => {
-      if(err) socket.emit('User not authenticated');
-      socket.emit('userdata', data);
-      // handles socket events
+  let user_id = '';
+
+  socket.on('authenticate', (_, clientCallback) => {
```

Comment:
Sorry if I am being think, my brain is kind of broken now but i am confused. Surely this is not the api for `socket.on`: https://socket.io/docs/server-api/#socket-on-eventname-callback